OneCommunityGlobal/HighestGoodNetworkApp
# Aaron - Add new field "Start Date" to Create New User modal

# Description
![image](https://github.com/OneCommunityGlobal/HGNRest/assets/10265513/7ec0cc3b-a44c-4ab6-a15a-4a06e8a59f0b)

- the PR implements all the highlighted functionality (the red was discussed with Jae and it's no longer necessary)

## Related PRS (if any):
backend PR is [PR472](https://github.com/OneCommunityGlobal/HGNRest/pull/472)

## Main changes explained:
- when creating a new user as an admin/owner, creator can now choose the start date (uses date picker)
- date picker defaulted to the next day
- the first time a user logs time into HGN, their start date is updated to the current day 

## How to test:
1. check into current branch and backend branch with the same name
2. do `npm install` and `npm run start:local` to run this PR locally
3. log in as admin user
4. go to Other Links → User Management and click Create New User button
5. fill in whatever credentials you want except for Start Date
6. play around with the date picker for Start Date to make sure it works (switching months, switching days updates the field)
7. pick a date that's not today and click Create

Log out and log into the account you just created

1. Go to your profile and verify the start date is the date you set
2. Log time into HGN
3. Go back to your profile and verify the start date has been updated

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/10265513/7f002589-790a-4910-8b50-8c1259c6b9ff

## Note:
- createdDate used UTC for timezone before this change, so I went with the same convention and that's why my start date in the video is offset by 1 day from the timelog's date 
- to test if the start date only updates the first time, you'd have to wait one day and log time during the next day and make sure the start date didn't change. A workaround for this is adding a console statement in line 484 of `TimeEntryForm.jsx` and verify it's not printed for subsequent time logs
